types/vote_set.go

```diff
@@ -671,6 +672,110 @@ func (voteSet *VoteSet) MakeExtendedCommit(fp FeatureParams) *ExtendedCommit {
 	return ec
 }
 
+// MakeBLSCommit is a variant of MakeExtendedCommit that aggregates BLS signatures.
+//
+// It additionally aggregates the BLS signatures for the block and nil. The
+// resulting commit contains only two aggregated signatures:
+//
+// 1 - aggregated signature for the block (compressed)
+// 2 - aggregated signature for nil (compressed).
```

Comment:
Not sure how useful it is to aggregate `nil` votes in practice (it's not on the happy path)

Reply:
Is it a question of usefulness? I thought it was more about the correctness of execution. Even if it's not on the happy path, if some validators voted for nil, we MUST verify their votes.